### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by ankitdimania-eligible**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2015 eligible
+Copyright (c) 2015 eligible.com
```
Comment (eligiblelaurence) [inline LICENSE]: Eligible Inc

File: .gitignore [hunk 2]
```diff
@@ -13,3 +13,13 @@ hs_err_pid*
 
 # Idea files
 .idea
+*.iml
+
+# Vim files
+*.swp
```
Comment (eligiblelaurence) [inline .gitignore]: i usually do *.sw[a-z] for safety as the file names change the more swapfiles you get

File: src/main/java/com/eligible/net/LiveEligibleResponseGetter.java [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+                    "fetch", requestClass);
+            fetchMethod.setAccessible(true);
+            Object response = fetchMethod.invoke(urlFetchService, request);
+
+            int responseCode = (Integer) response.getClass()
+                    .getDeclaredMethod("getResponseCode").invoke(response);
+            String body = new String((byte[]) response.getClass()
+                    .getDeclaredMethod("getContent").invoke(response), APIResource.CHARSET);
+            return new EligibleResponse(responseCode, body);
+        } catch (InvocationTargetException e) {
```
Comment (eligiblelaurence) [inline src/main/java/com/eligible/net/LiveEligibleResponseGetter.java]: Use a subclass to catch all of these

File: src/main/java/com/eligible/net/RequestType.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.eligible.net;
+
+/**
+ * Created by ankitdimania on 22/09/15.
```
Comment (eligiblelaurence) [inline src/main/java/com/eligible/net/RequestType.java]: it'll be in blame logs

Comment (eligiblelaurence) [inline src/main/java/com/eligible/net/RequestType.java]: if you want your name on it add a CONTRIBUTORS file or have a standard header

**Comment by eligiblelaurence** [comment]
Wow, this is awesome. So thorough

File: src/main/java/com/eligible/net/RequestType.java (hunk 4 above)
Comment (ankitdimania-eligible) [inline src/main/java/com/eligible/net/RequestType.java]: default template put these comments while creating a new file. I'll have to add documentation, so I'll get rid of all this.

File: .gitignore (hunk 2 above)
Comment (ankitdimania-eligible) [inline .gitignore]: :+1:

File: LICENSE (hunk 1 above)
Comment (ankitdimania-eligible) [inline LICENSE]: done

**Pull request merged**